Src/VimCore/Vim.fs

```diff
@@ -154,6 +155,9 @@ type internal VimData(_globalSettings: IVimGlobalSettings) as this =
         member x.LastSubstituteData 
             with get() = _lastSubstituteData
             and set value = _lastSubstituteData <- value
+        member x.LastLineCommand 

```

Comment:
Why can't `LastCommandLine` be used here vs. adding the new property?

Reply:
Because they're different types ... Need to finish coffee before I review code. :( 

---
In reply to: [189437792](https://github.com/jaredpar/VsVim/pull/2141#discussion_r189437792) [](ancestors = 189437792)

Reply:
The main reason we cannot use a string as the last command is that we need to support repeating the repeat. For example, `:s/pattern/replacement/`, then `:@:`, and then `:@:` again. For the second repeat, LastCommandLine is `":@:"`, but we want to repeat the substitute.

A secondary reason is that we need to support merging a new line range with a previous line command. If the last command line is a string, we would have to partially reparse it.